README.md

```diff
@@ -234,32 +276,41 @@ metadata:
   displayName: string # optional
 spec:
   thresholdMetric: # either thresholdMetric or ratioMetric should be provided
-    source: string # data source for the metric
-    queryType: string # a name for the type of query to run on the data source
-    query: string # the query to run to return the metric
-    metadata: # optional, allows data source specific details to be passed
+    metricSource:
+      metricSourceRef: string # optional, this field can be used to refer to DataSource object
+      type: string # optional, this field describes predefined metric source type e.g. Prometheus, Datadog, etc.
+      spec:
+        # arbitrary chosen fields for every data source type to make it comfortable to use
+        # anything that is valid YAML can be put here, user can refer to a particular
+        # DataSource, so that there are no connection specific details in the SLI
```

Comment:
Should this comment mention referring to a `DataSource`, or should it be removed as `metricSourceRef` handles the reference?

Reply:
I believe that the goal of this comment was to point out that `spec` might be used to hold some of the details about the connection, but I agree that this information is redundant. I will remove it :)

Reply:
I believe that the goal of this comment was to point out that `spec` might be used to hold some of the details instead of a `DataSource`, but I agree that this information is redundant. I will remove it :)